### PR TITLE
Handle wrong network connection

### LINKED ISSUE #230: "Add to cart" button shows "No wallet found" modal
https://www.notion.so/efdn/Add-to-cart-button-shows-No-wallet-found-modal-0b1ea297c1e84b539d2b4c90c1bd9be6

I'm not yet connected to the app but I'm already logged into MetaMask. When I click the "Add to cart" button on a project card, I see this modal:

Image 2021-08-05 at 8 56 36 AM

This is a new bug, I suspect this was introduced in the WalletConnect integration (#196).

### LINKED ISSUE #231: Specify which network to connect to in the wallet modal
https://www.notion.so/efdn/Specify-which-network-to-connect-to-in-the-wallet-modal-a4806dfd4bec4e8cbf4723944ba84d7d

1. I'm logged into MetaMask on Mainnet
2. I go to https://eth2clrfund.netlify.app/#/projects
3. I click "Connect"
4. I click MetaMask option
5. I sign the Signature Request modal
6. I see this error:
Image 2021-08-05 at 12 05 36 PM

A few things we could improve:
- [ ] Update the error copy to specify which network to connect to, i.e. "Wrong Network. Please connect to the appropriate Ethereum network." --> "Wrong Network. Please connect to the [network_name] Ethereum network."
- [ ] If we can already detect a provider before step 4 (i.e. before they make a selection), we should display that warning so the user avoids having to go through step 4 & 5.

### PR DESCRIPTION
**Pull request by pettinarip**
Fix #231
Fix #230

Things addressed here:
- Clean the code in order to only have two connection states: you can be connected or disconnected. E.g. you can not be connected but with the wrong network, in this case, you will be just disconnected.
- When the user is connected and he/she changes the network manually, the app will automatically disconnect (this is how it was working before).
- `ConnectionModal` was removed. It was only being used in the `AddToCartButton`. Now, when you click on that button it will display the `WalletModal` directly (if you are not connected) to pick whatever wallet you want.
- When the user tries to connect and he/she is using an unsupported network, the connection will fail and the error message will tell the user which network is supported.

### CODE REVIEW

File: vue-app/src/components/AddToCartButton.vue [hunk 1]
```diff
@@ -115,16 +115,16 @@ export default class AddToCartButton extends Vue {
 
   promptConnection(): void {
     return this.$modal.show(
-      ConnectionModal,
-      {},
+      WalletModal,
       {},
+      { width: 400, top: 20 },
```
Comment (pettinarip) [inline vue-app/src/components/AddToCartButton.vue:120]: The same sizes we are using in the `WalletWidget`.

File: vue-app/src/components/Cart.vue [hunk 2]
```diff
@@ -409,6 +410,18 @@ export default class Cart extends Vue {
...
+    return Number(process.env.VUE_APP_ETHEREUM_API_CHAINID)
+  }
+
+  get networkName(): string {
+    return CHAIN_INFO[this.supportedChainId].label
+  }
+
+  isCorrectNetwork(): boolean {
+    return this.supportedChainId === this.walletChainId
+  }
```
Comment (pettinarip) [inline vue-app/src/components/Cart.vue:425]: I simplify a bit this logic. There was no need to check into the "default" `jsonRpcNetwork`. We can check the supported chainId by reading the env var `VUE_APP_ETHEREUM_API_CHAINID`.

File: vue-app/src/components/WalletModal.vue [hunk 3]
```diff
@@ -85,9 +82,6 @@ export default class WalletModal extends Vue {
   async loginUser(user: User): Promise<void> {
     this.$emit('connected')
 
-    const url = await getProfileImageUrl(user.walletAddress)
-    this.updateProfileImage(url)
```
Comment (pettinarip) [inline vue-app/src/components/WalletModal.vue:89]: Removed this "task" from here (there was no reason to let the `loginUser` take care of that).

Now the consumers fetch the image when the user is connected. We are doing this in the `WalletWidget` by watching the `currentUser` object.

File: vue-app/src/components/WalletWidget.vue [hunk 4]
```diff
@@ -1,11 +1,7 @@
 <template>
   <div :class="{ container: !isActionButton }">
-    <template v-if="!isLoaded()"></template>
-    <div v-if="walletProvider && !isCorrectNetwork()" class="provider-error">
-      Please change network to {{ networkName }}
-    </div>
```
Comment (pettinarip) [inline vue-app/src/components/WalletWidget.vue:6]: We are not going to face this scenario anymore. You will be connected or disconnected.

As mentioned in the description, if the user changes the network, the app is going to automatically disconnect him.

**Review by corwintines — APPROVED** [review]
Nice cleanup of wallet stuff

**Review by samajammin — APPROVED** [review]
Nicely done!

**Review by tyntyn1 — COMMENTED** [review]
@pettinarip

**Pull request merged**